inference-engine/thirdparty/clDNN/tests/test_cases/fusings_gpu_test.cpp

```diff
@@ -8114,21 +8113,88 @@ class ScatterNDUpdatePrimitiveFusingTest : public ::BaseFusingTest<scatter_nd_up
     layout get_per_channel_layout(scatter_nd_update_test_params& p) {
         return layout{ p.default_type, p.default_format, tensor{1, p.input_shape.feature[0], 1, 1} };
     }
+
+    format get_default_format(int rank = 4) {
+        if (rank <= 4)
+            return cldnn::format::bfyx;
+        else if (rank == 5)
+            return cldnn::format::bfzyx;
+        else
+            return cldnn::format::bfwzyx;
+    }
+
+    template<typename T>
+    T generate_random_val(int min, int max, int k = 8) {
+        static std::default_random_engine generator(random_seed);
+        // 1/k is the resolution of the floating point numbers
+        std::uniform_int_distribution<int> distribution(k * min, k * max);
+        T val = (T)distribution(generator);
+        val /= k;
+
+        return val;
+    }
+
+    template <typename T>
+    std::vector<T> generate_unique_indices(scatter_nd_update_test_params& p) {
+        std::set<std::vector<T>> unique_indices;
+        std::vector<T> result;
+        auto indices_shape = p.indices_shape.sizes(get_default_format(p.indices_rank));
+        auto last_indices_dim = indices_shape.back();
+
+        auto count = 1;
+        for (size_t i = 0; i < indices_shape.size() - 1; i++)
+            count *= indices_shape[i];
+
+        while (unique_indices.size() != count) {
+            std::vector<T> indices;
+            for (size_t i = 0; i < last_indices_dim; i++)
+                indices.push_back(generate_random_val<T>(0, indices_shape[i]));
+
+            unique_indices.insert(indices);
+        }
+
+        std::for_each(unique_indices.begin(),
+                      unique_indices.end(),
+                      [&](const std::vector<T>& indices) {
+                          result.insert(result.end(), indices.begin(), indices.end());
+                      });
+
+        return result;
+    }
+
+    cldnn::memory::ptr get_indices_mem(scatter_nd_update_test_params& p) {
```

Comment:
can't `get_repeatless_mem` be used instead?

Reply:
It can't be used since we need not a just separate repeatless values, but a sequence of values related to dims indices. For example if `last_indices_dim == 4` we need to generate 4 values {v1, v2, v3, v4} which will be used to address output data at `b=v1 f=v2 y=v3 x=v4` position. So, this sequence should be unique and address should be modified only once. Moreover we can't generate values bigger than related dimension size, which means the possible range for max value should be limited by smallest dim in case of `get_repeatless_mem`, which may not be enough

Reply:
And the number of such tuples (consisting of 4 elements as for example above) is `indices_shape[:-1]`